python_modules/dagster/dagster_tests/core_tests/graph_tests/test_op_selection.py

```diff
@@ -331,3 +335,289 @@ def test_disconnected_selection():
+    assert result_graph.success
+    assert set(_success_step_keys(result_graph)) == {
+        "subgraph.return_one",
+    }
+    assert result_graph.output_for_node("subgraph.return_one") == 1
+
+
+def test_nested_graph_selection_inside_graph():
+    @graph
+    def subgraph():
+        return add_one(adder(return_one(), return_two()))
+
+    @job
+    def _super():
+        add_one(subgraph())
+
+    # select inside subgraph
+    result_graph = _super.execute_in_process(
+        op_selection=[
+            "subgraph.return_one",
+            "subgraph.return_two",
+            "subgraph.adder",
+        ]
+    )
+    assert result_graph.success
+    assert set(_success_step_keys(result_graph)) == {
+        "subgraph.return_one",
+        "subgraph.return_two",
+        "subgraph.adder",
+    }
+    assert result_graph.output_for_node("subgraph.adder") == 3
+
+
+def test_nested_graph_selection_both_inside_and_outside_disconnected():
+    @graph
+    def subgraph():
+        return add_one(adder(return_one(), return_two()))
+
+    @job
+    def _super():
+        add_one(subgraph())
+
+    # select inside subgraph (disconnected to the outside) and outside
+    with pytest.raises(DagsterInvalidSubsetError):
+        # can't build graph bc "subgraph" won't have output mapping but "add_one" expect output
+        _super.execute_in_process(
+            op_selection=["subgraph.adder", "add_one"],
+            run_config={
+                "ops": {"subgraph": {"ops": {"adder": {"inputs": {"num1": 10, "num2": 20}}}}}
+            },
+        )
+
+
+def test_nested_graph_selection_unsatisfied_subgraph_inputs():
+    @graph
+    def subgraph():
+        return add_one(adder(return_one(), return_two()))
+
+    @job
+    def _super():
+        add_one(subgraph())
+
+    # output mapping
+    result_sub_1 = _super.execute_in_process(
+        # TODO query not working yet
+        # op_selection=["subgraph.return_one*"],
+        op_selection=["subgraph.return_one", "subgraph.adder", "subgraph.add_one", "add_one"],
+        run_config={"ops": {"subgraph": {"ops": {"adder": {"inputs": {"num2": 0}}}}}},
+    )
+    assert result_sub_1.success
+    assert set(_success_step_keys(result_sub_1)) == {
+        "subgraph.return_one",
+        "subgraph.adder",
+        "subgraph.add_one",
+        "add_one",
+    }
+    assert result_sub_1.output_for_node("add_one") == 3
+
+    # no output mapping
+    result_sub_2 = _super.execute_in_process(
+        op_selection=["subgraph.return_one", "subgraph.adder", "subgraph.add_one"],
+        run_config={"ops": {"subgraph": {"ops": {"adder": {"inputs": {"num2": 100}}}}}},
+    )
+    assert result_sub_2.success
+    assert set(_success_step_keys(result_sub_2)) == {
+        "subgraph.return_one",
+        "subgraph.adder",
+        "subgraph.add_one",
+    }
+    assert result_sub_2.output_for_node("subgraph.add_one") == 102
+
+
+def test_nested_graph_selection_input_mapping():
```

Comment:
what about the case where you select a nested graph w/input mapping without selecting upstream? You should still be able to provide input via config right?

Reply:
yup!